lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:collection';
+import 'dart:convert';
+
+class Token {
+  String token;
+  int? position;
+  final int line;
```

Comment:
Could we use `SourceSpan` from: https://pub.dev/packages/source_span to represent the position?

It's fine if we depend on `package:source_span` it's made by `tools.dart.dev` and used in the Dart SDK / analyzer and such :D 


Also consider making these fields final, so that they can't be modified after an instance of `Token` has been created.
```dart
class Token {
  final String token;
  final SourceSpan span;
}

Reply:
I was not making it final as I was updating their text while cleaning up tokens, but I can always a new instance of token add it in a separate list. I'll make them final.

Reply:
I'll try with source span and will let you know if I have any problems.